src/routes/github/github-oauth.ts

```diff
@@ -33,37 +41,45 @@ const getRedirectUrl = async (res, state) => {
 	return `${hostname}/login/oauth/authorize?client_id=${clientId}&scope=${encodeURIComponent(scopes.join(" "))}&redirect_uri=${encodeURIComponent(callbackURI)}&state=${state}`;
 };
 
-const GithubOAuthLoginGet = async (req: Request, res: Response): Promise<void> => {
+export const GithubOAuthLoginGet = async (req: Request, res: Response): Promise<void> => {
 	// TODO: We really should be using an Auth library for this, like @octokit/github-auth
 	// Create unique state for each oauth request
-	const state = crypto.randomBytes(8).toString("hex");
+	const stateKey = crypto.randomBytes(8).toString("hex");
 
 	req.session["timestamp_before_oauth"] = Date.now();
 
 	const parsedOriginalUrl = url.parse(req.originalUrl);
 	// Save the redirect that may have been specified earlier into session to be retrieved later
-	req.session[state] =
-		res.locals.redirect ||
-		`/github/configuration${parsedOriginalUrl.search || ""}`;
+
+	const state: OAuthState = {
+		postLoginRedirectUrl: res.locals.redirect ||
+		`/github/configuration${parsedOriginalUrl.search || ""}`,
+		installationIdPk: res.locals.installation.id,
+		gitHubServerUuid: res.locals.gitHubAppConfig.uuid,
```

Comment:
Base on the gitHubRouter comments, we may not even need this? As the uuid should be in the callback url?

Reply:
We cannot trust the URL without validation, however that's a good point! I don't think we need UUID for the callback whatsoever. Will try to remove UUID from that particular URL.